client/src/app/App.js

```diff
@@ -75,6 +75,8 @@ export const EMPTY_TAB = {
 
 const ENCODING_UTF8 = 'utf8';
 
+const UUID_CONFIG_KEY = 'instance_uuid';
```

Comment:
A few things to consider:

* It's technically not an instance UUID but a per-machine user / editor id => Maybe we can simply rename it to `uuid` or `editor_id` or `session_id`?
* Generate in client or back-end?
* Should we store that in the config or next to it? Maybe users share their configs and I'd rather not like that `uuid` to be shared. => Idea could be put it in a separate file so it is not accidentally shared.

Depending on how much we rely on the ID in the future, I'd rather ensure it is there on the server side, as early as possible. Example: App cannot be started => Crash Report => UUID generated?

Reply:
You're right about the naming, I'll convert it to editor_id. I think it is semantically more correct.

For the UUID generation, IMHO handling it on the backend would just make things more complicated and does not provide a better uniqueness (unless we keep all the generated UUID somewhere and whenever we generate a new one, we compare the new one with the old ones. That'd also mean increasing memory usage by time if we don't store the generated UUIDs in a DB or something.) Based on the algorithm and modern Math.random implementations, the UUID generation algorithm we use within this PR gives us enough uniqueness.

I'd say if the app crashes for some reason before the UUID is generated, let's just send a request with null UUID so that the backend would know the UUID could not be generated. We'd still be sending the crash report which would provide us a meaningful log to understand the bug anyway. 

So I personally think UUID generation in the backend would not provide us any additional functionality (but maybe I'm missing a point). If you think that's the way to go, we can go that way though.

For the UUID storage, so we wanna have a separate file just like the config file to store the UUID right? That'd make sense, does not provide any more security, etc. but yep it would not make sense if the users are sharing their configs and it also contains the UUID.

Reply:
> So I personally think UUID generation in the backend would not provide us any additional functionality (but maybe I'm missing a point). If you think that's the way to go, we can go that way though.

Please checkout our configuration provider on the app side, cf. https://github.com/camunda/camunda-modeler/blob/develop/app/lib/config/index.js.

Adding a new provider that generates the UUID and provides it to the client could be straight forward. If we decide to save it in a separate file, we should make sure that we keep the actual behavior contained in a single place (and do not blow up our existing API surfaces).

> For the UUID generation, IMHO handling it on the backend would just make things more complicated and does not provide a better uniqueness (unless we keep all the generated UUID somewhere and whenever we generate a new one, we compare the new one with the old ones. That'd also mean increasing memory usage by time if we don't store the generated UUIDs in a DB or something.) Based on the algorithm and modern Math.random implementations, the UUID generation algorithm we use within this PR gives us enough uniqueness.

I do not understand this one. Generating a UUID in the backend is as unique as generating it on the client. For me it is just a mattern _when_ to generate it and I believe doing that early rather than on editor window open is the right way to go.


> You're right about the naming, I'll convert it to editor_id. I think it is semantically more correct.

:+1:

Reply:
Ah, now I got your point. You mean the Electron Backend. I thought you were talking about the update server. Then yes, it would make sense.

Reply:
Good we sorted that one out :speak_no_evil:

Reply:
The Camunda Modeler is a client - server application. You'll regularly hear me talking about client/front-end and back-end there.